src/librbd/librbd.cc

```diff
@@ -61,6 +62,16 @@ namespace {
 
 TracepointProvider::Traits tracepoint_traits("librbd_tp.so", "rbd_tracing");
 
+buffer::raw* create_write_raw(librbd::ImageCtx *ictx, const char *buf,
+                              size_t len) {
+  // cannot allow the cache (if enabled) to retain reference to user memory
+  if (ictx->cache) {
+    return buffer::copy(buf, len);
+  } else {
+    return buffer::create_static(len, const_cast<char*>(buf));
```

Comment:
Yes, I hope we can support cancel_message to do this... Currently messenger exposes too little